Fix example in `OS.shell_open()` method documentation and add a new one

mateuseap opened the pull request:
Closes [#10367](https://github.com/godotengine/godot-docs/issues/10453)

### What I Did:
- Fixed an issue with string escaping in an example provided for the `OS.shell_open()` method documentation. Now the example is properly escaped to reflect valid Windows path syntax.
- Added a new example of how to open a Windows path using the `OS.shell_open()` method with a different syntax from the first Windows path example.

### Demonstrations:
- Current Windows path example shown in the `OS.shell_open()` documentation displaying error:
   ![image](https://github.com/user-attachments/assets/7b43d50c-d79f-41dd-b0d3-d2e2ee24ab61)
- Corrected Windows path example showing no errors:
   ![image](https://github.com/user-attachments/assets/ba98cb62-50be-4160-8e32-ef4890f6e1e2)
   - Note: I've tested this GDScript, and it opens the folder as expected. I attached the script to a node and ran the scene, and it worked perfectly. If needed, I can provide a GIF.

RedMser commented on doc/classes/OS.xml:
```diff
@@ -809,7 +809,7 @@
 			<description>
 				Requests the OS to open a resource identified by [param uri] with the most appropriate program. For example:
-				- [code]OS.shell_open("C:\\Users\name\Downloads")[/code] on Windows opens the file explorer at the user's Downloads folder.
+				- [code]OS.shell_open("C:\\\\Users\\name\\Downloads")[/code] on Windows opens the file explorer at the user's Downloads folder.
```
I haven't used this method before but fairly sure there's too many backslashes here at the start:

```suggestion
				- [code]OS.shell_open("C:\\Users\\name\\Downloads")[/code] on Windows opens the file explorer at the user's Downloads folder.
```

mateuseap replied:
Indeed, you're absolutely right, @RedMser. It doesn’t even make sense to have so many backslashes. I've corrected this and also added a updated the PR description.

bruvzg reviewed (approved):
Note: `shell_open` on Windows will also work with `/`, so `[code]OS.shell_open("C:/Users/name/Downloads")[/code]` can be added as alternative example as well.

Mickeon commented:
I think we could just stick to using `/` but I have no strong opinion on it. Windows is the odd one out.

mateuseap committed:
Fix incorrect string escaping in OS.shell_open() method Windows path example

Corrected improper string escaping in the OS.shell_open() Windows path example. Backslashes are now properly escaped to prevent string parsing errors. Also, added a new alternative example of how to mount a Windows path with OS.shell_open().

mateuseap commented:
@bruvzg I've added a new example using `/` only, as you suggested!
